src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpContentStream.cs

```diff
@@ -97,5 +98,27 @@ protected static void ValidateCopyToArgs(Stream source, Stream destination, int
                     (Exception)new ObjectDisposedException(nameof(destination), SR.ObjectDisposed_StreamClosed);
             }
         }
+
+        /// <summary>Gets a <see cref="Task{int}"/> for a <see cref="ValueTask{int}"/>.</summary>
+        /// <param name="valueTask"></param>
+        protected static Task<int> AsTask(ValueTask<int> valueTask)
```

Comment:
Could this just be part of ValueTask?

Reply:
In theory, yes.  In practice, I'm not sure we want every app to pay the costs of this cache.  ValueTask.AsTask() in 2.1 already uses the same task cache that synchronously completing async methods do, which for ints mean there's a cache of IIRC the values [-1, 8].  We could:
a) Undo this change I made
b) Keep this change
c) Keep it and change the value
d) Undo this change and modify the AsyncTaskCache in coreclr to use a higher bound than 8 inclusive.

Thinking more about it, we should probably do (a) or (d).  I'll go with (a) for now and start a discussion about (d).